combine.lua

```diff
@@ -689,6 +701,11 @@ local function merge_stacks(stacks)
                 if item.after_size == 0 then
                     log(4, '      removing')
                     dfhack.items.remove(item.item)
+                    if dfhack.world.isAdventureMode() and df.global.game.main_interface.adventure.inventory.open then
+                        -- refresh the inventory to prevent stale item references
+                        print('!! closed adventure mode inventory screen to prevent stale item references !!')
+                        df.global.game.main_interface.adventure.inventory.open = false
```

Comment:
why can't we send LEAVESCREEN here? If we're in a "details" screen, we can send LEAVESCREEN twice, no?

Reply:
How do we do that?

Reply:
`gui.simulateInput(dfhack.gui.getDFViewscreen(), 'LEAVESCREEN')`
and then if you're not back at the default adventurer view (use `dfhack.gui.matchFocusString`), then send it again